### PR TITLE
Prepare for ccm

### PR DESCRIPTION
**Pull request by boaks**
Fix some logging statements.
Fix some misleading comments.
Use dtls_decrypt/encrypt with params to prepare for CCM additional to CCM8.

Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

**Comment by boaks** [comment]
This seems to be also a easy one.

File: dtls.c [hunk 1]
```diff
@@ -1496,10 +1496,14 @@ dtls_prepare_record(dtls_peer_t *peer, dtls_security_parameters_t *security,
     memcpy(A_DATA + 8,  &DTLS_RECORD_HEADER(sendbuf)->content_type, 3); /* type and version */
     dtls_int_to_uint16(A_DATA + 11, res - 8); /* length */
 
-    res = dtls_encrypt(start + 8, res - 8, start + 8, nonce,
-		       dtls_kb_local_write_key(security, peer->role),
-		       dtls_kb_key_size(security, peer->role),
-		       A_DATA, A_DATA_LEN);
+    /* For backwards-compatibility, dtls_encrypt_params is called with
+     * M=<macLen> and L=3. */
+    const dtls_ccm_params_t params = { nonce, 8, 3 };
```
Comment (obgm) [inline dtls.c:1501]: To be backwards compatible to really old compilers, we have so far put variable definitions on the beginning of a block. `params` therefore should be moved to the beginning of the function body or enclosed together with the `dtls_encrypt_params()` invocation in `{` ... `}`.
The same holds below in `decrypt_verify()`.

**Review by obgm — COMMENTED** [review]
I have a small comment, otherwise it is ready for merge.

File: dtls.c (hunk 1 above)
Comment (boaks) [reply dtls.c:1501]: OK, I adapt it.

Comment (boaks) [reply dtls.c]: The params are move. I removed the const as well, though the intention would be to place the macLength there.

Comment (boaks) [reply dtls.c]: if it's OK for you, I squash the commits.

Comment (boaks) [reply dtls.c]: I also adapted the "add_ccm" PR. It's possible to get the macLen ahead. so const will also work. I will change it again.

Comment (obgm) [reply dtls.c]: Ok, great. Please squash.

**Comment by obgm** [comment]
When squashing, may I ask you to pick a more specific first line in the commit message that describes what this commit does in detail (prefixed by the name of the affected file: `dtls.c:`.

**Comment by boaks** [comment]
sure ...

**Comment by boaks** [comment]
> dtls.c: Minor fixes in logging and comments.

?

**Comment by obgm** [comment]
Well, the major functional change here is the switch from `dtls_encrypt()` to `dtls_encrypt_params()`, IMO.

**Comment by boaks** [comment]
OK, next try :-).

**Comment by obgm** [comment]
Great, thank you very much!

**Pull request merged**